src/Components/AddItem.jsx

```diff
@@ -1,5 +1,76 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import firebase from 'firebase';
+import { db } from '../lib/firebase';
 
 export default function AddItem() {
-  return <div>Hi from Add an Item View!</div>;
+  const [token, setToken] = useState('test-token');
+  const { register, handleSubmit, watch, errors } = useForm();
+  const onSubmit = (data) => {
+    const currentList = db.collection('lists').doc(token);
+    currentList
+      .update({
+        items: firebase.firestore.FieldValue.arrayUnion({
+          name: data.itemName,
+          frequency: parseInt(data.itemFrequency),
+          lastPurchased: null,
+        }),
+      })
+      .then(() => alert('Item has been submitted!'));
```

Comment:
This looks great! We think it would be good to have the data reset after a successful submission. [This link](https://react-hook-form.com/api/#reset) looks similar to the current form code and may be helpful to get that working.

Reply:
Thanks for the suggestion! We just added this functionality.

Reply:
Everything looks great now. Good job!